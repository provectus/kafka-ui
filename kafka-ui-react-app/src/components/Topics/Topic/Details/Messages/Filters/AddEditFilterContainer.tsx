import React from 'react';
import * as S from 'components/Topics/Topic/Details/Messages/Filters/Filters.styled';
import { InputLabel } from 'components/common/Input/InputLabel.styled';
import Input from 'components/common/Input/Input';
import { Textarea } from 'components/common/Textbox/Textarea.styled';
import { FormProvider, Controller, useForm } from 'react-hook-form';
import { ErrorMessage } from '@hookform/error-message';
import { Button } from 'components/common/Button/Button';
import { FormError } from 'components/common/Input/Input.styled';
import { MessageFilters } from 'components/Topics/Topic/Details/Messages/Filters/Filters';
import { yupResolver } from '@hookform/resolvers/yup';
import yup from 'lib/yupExtended';

const validationSchema = yup.object().shape({
  name: yup.string().required(),
  code: yup.string().required(),
});

export interface AddEditFilterContainerProps {
  title: string;
  cancelBtnHandler: () => void;
  submitBtnText: string;
  inputDisplayNameDefaultValue?: string;
  inputCodeDefaultValue?: string;
  toggleSaveFilterValue?: boolean;
  toggleSaveFilterSetter?: () => void;
  createNewFilterText?: string;
  submitCallback?: (values: MessageFilters) => void;
  submitCallbackWithReset?: boolean;
}

const AddEditFilterContainer: React.FC<AddEditFilterContainerProps> = ({
  title,
  cancelBtnHandler,
  submitBtnText,
  inputDisplayNameDefaultValue = '',
  inputCodeDefaultValue = '',
  toggleSaveFilterValue,
  toggleSaveFilterSetter,
  createNewFilterText,
  submitCallback,
  submitCallbackWithReset,
}) => {
  const methods = useForm<MessageFilters>({
    mode: 'onChange',
    resolver: yupResolver(validationSchema),
  });
  const {
    handleSubmit,
    control,
    formState: { isDirty, isSubmitting, isValid, errors },
    reset,
  } = methods;

  const onSubmit = React.useCallback(
    (values: MessageFilters) => {
      submitCallback?.(values);
      if (submitCallbackWithReset) {
        reset({ name: '', code: '' });
      }
    },
    [reset, submitCallback, submitCallbackWithReset]
  );

  return (
    <>
      <S.FilterTitle>{title}</S.FilterTitle>
      <FormProvider {...methods}>
        {createNewFilterText && (
          <S.CreatedFilter>{createNewFilterText}</S.CreatedFilter>
        )}
        <form
          onSubmit={handleSubmit(onSubmit)}
          aria-label="Filters submit Form"
        >
          <div>
            <InputLabel>Display name</InputLabel>
            <Input
              inputSize="M"
              placeholder="Enter Name"
              autoComplete="off"
              name="name"
              defaultValue={inputDisplayNameDefaultValue}
            />
          </div>
          <div>
            <FormError>
              <ErrorMessage errors={errors} name="name" />
            </FormError>
          </div>
          <div>
            <InputLabel>Filter code</InputLabel>
            <Controller
              control={control}
              name="code"
              defaultValue={inputCodeDefaultValue}
              render={({ field: { onChange, ref } }) => (
                <Textarea ref={ref} onChange={onChange} />
              )}
            />
          </div>
          <div>
            <FormError>
              <ErrorMessage errors={errors} name="code" />
            </FormError>
          </div>
          {!!toggleSaveFilterSetter && (
            <S.CheckboxWrapper>
              <input
                type="checkbox"
                checked={toggleSaveFilterValue}
                onChange={toggleSaveFilterSetter}
              />
              <InputLabel>Save this filter</InputLabel>
            </S.CheckboxWrapper>
          )}
          <S.FilterButtonWrapper>
            <Button
              buttonSize="M"
              buttonType="secondary"
              type="button"
              onClick={cancelBtnHandler}
            >
              Cancel
            </Button>
            <Button
              buttonSize="M"
              buttonType="primary"
              type="submit"
              disabled={!isValid || isSubmitting || !isDirty}
            >
              {submitBtnText}
            </Button>
          </S.FilterButtonWrapper>
        </form>
      </FormProvider>
    </>
  );
};

export default AddEditFilterContainer;
