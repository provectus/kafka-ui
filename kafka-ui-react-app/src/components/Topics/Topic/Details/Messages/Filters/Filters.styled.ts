import { styled } from 'lib/themedStyles';
import { Colors } from 'theme/theme';

export const FiltersWrapper = styled.div`
  display: flex;
  flex-direction: column;
  padding-left: 16px;
  padding-right: 16px;

  & > div:first-child {
    display: flex;
    justify-content: space-between;
    padding-top: 16px;

    & > div:last-child {
      width: 10%;
    }
  }

  & .multi-select {
    height: 32px;
    & > .dropdown-container {
      height: 32px;
      & > .dropdown-heading {
        height: 32px;
      }
    }
  }
`;

export const FilterInputs = styled.div`
  display: flex;
  gap: 8px;
  align-items: center;
  width: 90%;

  & > div:first-child {
    width: 25%;
  }
`;

export const SeekTypeSelectorWrapper = styled.div`
  display: flex;
  & .select-wrapper {
    width: 40% !important;
    & > select {
      border-radius: 4px 0 0 4px !important;
    }
  }

  & .offset-selector {
    border-radius: 0 4px 4px 0 !important;
    border-left: none;
  }

  & .date-picker {
    height: 32px;
    border: 1px ${(props) => props.theme.selectStyles.borderColor.normal} solid;
    border-left: none;
    border-radius: 0 4px 4px 0;
    font-size: 14px;
    width: 100%;
    padding-left: 12px;
    color: ${(props) => props.theme.selectStyles.color.normal};

    background-image: url('data:image/svg+xml,%3Csvg width="10" height="6" viewBox="0 0 10 6" fill="none" xmlns="http://www.w3.org/2000/svg"%3E%3Cpath d="M1 1L5 5L9 1" stroke="%23454F54"/%3E%3C/svg%3E%0A') !important;
    background-repeat: no-repeat !important;
    background-position-x: 96% !important;
    background-position-y: 55% !important;
    appearance: none !important;

    &:hover {
      cursor: pointer;
    }
    &:focus {
      outline: none;
    }
  }
`;

export const FiltersMetrics = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
  gap: 22px;
  padding-top: 16px;
  padding-bottom: 16px;
`;

export const Metric = styled.div`
  color: ${Colors.neutral[50]};
  font-size: 12px;
  display: flex;
`;

export const MetricsIcon = styled.div`
  color: ${Colors.neutral[90]};
  padding-right: 6px;
  height: 12px;
`;
