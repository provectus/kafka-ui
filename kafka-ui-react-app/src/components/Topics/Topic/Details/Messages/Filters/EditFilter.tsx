import React from 'react';
import { MessageFilters } from 'components/Topics/Topic/Details/Messages/Filters/Filters';
import { FilterEdit } from 'components/Topics/Topic/Details/Messages/Filters/FilterModal';

import AddEditFilterContainer from './AddEditFilterContainer';

export interface EditFilterProps {
  editFilter: FilterEdit;
  toggleEditModal(): void;
  editSavedFilter(filter: FilterEdit): void;
}

const EditFilter: React.FC<EditFilterProps> = ({
  editFilter,
  toggleEditModal,
  editSavedFilter,
}) => {
  const onSubmit = React.useCallback(
    (values: MessageFilters) => {
      editSavedFilter({ index: editFilter.index, filter: values });
      toggleEditModal();
    },
    [editSavedFilter, editFilter.index, toggleEditModal]
  );
  return (
    <AddEditFilterContainer
      title="Edit saved filter"
      cancelBtnHandler={() => toggleEditModal()}
      submitBtnText="Save"
      inputDisplayNameDefaultValue={editFilter.filter.name}
      inputCodeDefaultValue={editFilter.filter.code}
      submitCallback={onSubmit}
    />
  );
};

export default EditFilter;
