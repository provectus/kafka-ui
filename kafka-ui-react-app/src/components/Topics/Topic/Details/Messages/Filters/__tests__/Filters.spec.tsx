import React from 'react';
import Filters, {
  FiltersProps,
} from 'components/Topics/Topic/Details/Messages/Filters/Filters';
import { render } from 'lib/testHelpers';

const setupWrapper = (props?: Partial<FiltersProps>) => (
  <Filters
    clusterName="test-cluster"
    topicName="test-topic"
    partitions={[{ partition: 0, offsetMin: 0, offsetMax: 100 }]}
    meta={{}}
    isFetching={false}
    addMessage={jest.fn()}
    resetMessages={jest.fn()}
    updatePhase={jest.fn()}
    updateMeta={jest.fn()}
    setIsFetching={jest.fn()}
    {...props}
  />
);
describe('Filters component', () => {
  it('matches the snapshot', () => {
    const component = render(setupWrapper());
    expect(component.baseElement).toMatchSnapshot();
  });
  describe('when fetching', () => {
    it('matches the snapshot', () => {
      const component = render(setupWrapper({ isFetching: true }));
      expect(component.baseElement).toMatchSnapshot();
    });
  });
});
