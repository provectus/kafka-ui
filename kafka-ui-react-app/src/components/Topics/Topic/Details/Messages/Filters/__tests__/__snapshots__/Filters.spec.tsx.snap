// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters component matches the snapshot 1`] = `
.c9 {
  min-width: 200px;
  height: 32px;
  font-size: 14px;
}

.c9 > .dropdown-container {
  height: 32px;
}

.c9 > .dropdown-container > .dropdown-heading {
  height: 32px;
}

.c3 {
  position: absolute;
  top: 50%;
  line-height: 0;
  z-index: 1;
  left: 12px;
  right: unset;
  height: 11px;
  width: 11px;
  color: #454F54;
}

.c4 {
  border: 1px #ABB5BA solid;
  border-radius: 4px;
  height: 32px;
  width: 100%;
  padding-left: 36px;
  font-size: 14px;
}

.c4::-webkit-input-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c4::-moz-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c4:-ms-input-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c4::placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c4:hover {
  border-color: #73848C;
}

.c4:focus {
  outline: none;
  border-color: #454F54;
}

.c4:focus::-webkit-input-placeholder {
  color: transparent;
}

.c4:focus::-moz-placeholder {
  color: transparent;
}

.c4:focus:-ms-input-placeholder {
  color: transparent;
}

.c4:focus::placeholder {
  color: transparent;
}

.c4:disabled {
  color: #ABB5BA;
  border-color: #E3E6E8;
  cursor: not-allowed;
}

.c4:read-only {
  color: #171A1C;
  border: none;
  background-color: #F1F2F3;
  cursor: not-allowed;
}

.c4:-moz-read-only:focus::placeholder {
  color: #ABB5BA;
}

.c4:read-only:focus::placeholder {
  color: #ABB5BA;
}

.c8 {
  border: 1px #ABB5BA solid;
  border-radius: 4px;
  height: 32px;
  width: 100%;
  padding-left: 12px;
  font-size: 14px;
}

.c8::-webkit-input-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c8::-moz-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c8:-ms-input-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c8::placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c8:hover {
  border-color: #73848C;
}

.c8:focus {
  outline: none;
  border-color: #454F54;
}

.c8:focus::-webkit-input-placeholder {
  color: transparent;
}

.c8:focus::-moz-placeholder {
  color: transparent;
}

.c8:focus:-ms-input-placeholder {
  color: transparent;
}

.c8:focus::placeholder {
  color: transparent;
}

.c8:disabled {
  color: #ABB5BA;
  border-color: #E3E6E8;
  cursor: not-allowed;
}

.c8:read-only {
  color: #171A1C;
  border: none;
  background-color: #F1F2F3;
  cursor: not-allowed;
}

.c8:-moz-read-only:focus::placeholder {
  color: #ABB5BA;
}

.c8:read-only:focus::placeholder {
  color: #ABB5BA;
}

.c2 {
  position: relative;
}

.c6 {
  position: relative;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border: 1px #ABB5BA solid;
  border-radius: 4px;
  font-size: 14px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding-left: 12px;
  padding-right: 16px;
  color: #171A1C;
  min-width: 100px;
  background-image: url('data:image/svg+xml,%3Csvg width="10" height="6" viewBox="0 0 10 6" fill="none" xmlns="http://www.w3.org/2000/svg"%3E%3Cpath d="M1 1L5 5L9 1" stroke="%23454F54"/%3E%3C/svg%3E%0A') !important;
  background-repeat: no-repeat !important;
  background-position-x: calc(100% - 8px) !important;
  background-position-y: 55% !important;
}

.c6:hover {
  color: #171A1C;
  border-color: #73848C;
}

.c6:focus {
  outline: none;
  color: #171A1C;
  border-color: #454F54;
}

.c6:disabled {
  color: #ABB5BA;
  border-color: #E3E6E8;
  cursor: not-allowed;
}

.c11 {
  position: relative;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border: 1px #ABB5BA solid;
  border-radius: 4px;
  font-size: 14px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding-left: 12px;
  padding-right: 16px;
  color: #171A1C;
  min-width: 120px;
  background-image: url('data:image/svg+xml,%3Csvg width="10" height="6" viewBox="0 0 10 6" fill="none" xmlns="http://www.w3.org/2000/svg"%3E%3Cpath d="M1 1L5 5L9 1" stroke="%23454F54"/%3E%3C/svg%3E%0A') !important;
  background-repeat: no-repeat !important;
  background-position-x: calc(100% - 8px) !important;
  background-position-y: 55% !important;
}

.c11:hover {
  color: #171A1C;
  border-color: #73848C;
}

.c11:focus {
  outline: none;
  color: #171A1C;
  border-color: #454F54;
}

.c11:disabled {
  color: #ABB5BA;
  border-color: #E3E6E8;
  cursor: not-allowed;
}

.c7 {
  padding-right: 16px;
  list-style-position: inside;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #F1F2F3;
  color: #171A1C;
  font-size: 14px;
  font-weight: 500;
  height: 32px;
}

.c10:hover:enabled {
  background: #E3E6E8;
  color: #171A1C;
  cursor: pointer;
}

.c10:active:enabled {
  background: #D5DADD;
  color: #171A1C;
}

.c10:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c10 a {
  color: #FFFFFF;
}

.c10 i {
  margin-right: 7px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 16px;
  padding-right: 16px;
}

.c0 > div:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 16px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 90%;
}

.c1 > div:first-child {
  width: 25%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 .select-wrapper {
  width: 40% !important;
}

.c5 .select-wrapper > select {
  border-radius: 4px 0 0 4px !important;
}

.c5 .offset-selector {
  border-radius: 0 4px 4px 0 !important;
  border-left: none;
}

.c5 .date-picker {
  height: 32px;
  border: 1px #ABB5BA solid;
  border-left: none;
  border-radius: 0 4px 4px 0;
  font-size: 14px;
  width: 100%;
  padding-left: 12px;
  color: #171A1C;
  background-image: url('data:image/svg+xml,%3Csvg width="10" height="6" viewBox="0 0 10 6" fill="none" xmlns="http://www.w3.org/2000/svg"%3E%3Cpath d="M1 1L5 5L9 1" stroke="%23454F54"/%3E%3C/svg%3E%0A') !important;
  background-repeat: no-repeat !important;
  background-position-x: 96% !important;
  background-position-y: 55% !important;
  -webkit-appearance: none !important;
  -moz-appearance: none !important;
  appearance: none !important;
}

.c5 .date-picker:hover {
  cursor: pointer;
}

.c5 .date-picker:focus {
  outline: none;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 22px;
  padding-top: 16px;
  padding-bottom: 16px;
}

.c13 {
  color: #73848C;
  font-size: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c14 {
  color: #171A1C;
  padding-right: 6px;
  height: 12px;
}

<body>
  <div>
    <div
      class="c0"
    >
      <div>
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <i
              class="c3 fas fa-search"
            />
            <input
              class="c4 c2"
              id="searchText"
              placeholder="Search"
              type="text"
              value=""
            />
          </div>
          <div
            class="c5"
          >
            <div>
              <ul
                class="c6"
                id="selectSeekType"
                role="listbox"
              >
                <li
                  class="c7"
                  role="option"
                  tabindex="0"
                >
                  Offset
                </li>
              </ul>
            </div>
            <div
              class="c2 offset-selector"
            >
              <input
                class="c8 c2 offset-selector"
                id="offset"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rmsc c9"
          >
            <div
              aria-labelledby="Select partitions"
              aria-readonly="true"
              class="dropdown-container"
              tabindex="0"
            >
              <div
                class="dropdown-heading"
              >
                <div
                  class="dropdown-heading-value"
                >
                  <span>
                    All items are selected.
                  </span>
                </div>
                <button
                  aria-label="Clear Selected"
                  class="clear-selected-button"
                  type="button"
                >
                  <svg
                    class="dropdown-search-clear-icon gray"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-width="2"
                    width="24"
                  >
                    <line
                      x1="18"
                      x2="6"
                      y1="6"
                      y2="18"
                    />
                    <line
                      x1="6"
                      x2="18"
                      y1="6"
                      y2="18"
                    />
                  </svg>
                </button>
                <svg
                  class="dropdown-heading-dropdown-arrow gray"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-width="2"
                  width="24"
                >
                  <path
                    d="M6 9L12 15 18 9"
                  />
                </svg>
              </div>
            </div>
          </div>
          <button
            class="c10"
            style="font-weight: 500;"
            type="submit"
          >
            Submit
          </button>
        </div>
        <div>
          <ul
            class="c11"
            role="listbox"
          >
            <li
              class="c7"
              role="option"
              tabindex="0"
            >
              Oldest First
            </li>
          </ul>
        </div>
      </div>
      <div
        class="c12"
      >
        <p
          style="font-size: 14px;"
        />
        <div
          class="c13"
          title="Elapsed Time"
        >
          <div
            class="c14"
          >
            <i
              class="far fa-clock"
            />
          </div>
          <span>
            0
             ms
          </span>
        </div>
        <div
          class="c13"
          title="Bytes Consumed"
        >
          <div
            class="c14"
          >
            <i
              class="fas fa-arrow-down"
            />
          </div>
          <span>
            0Bytes
          </span>
        </div>
        <div
          class="c13"
          title="Messages Consumed"
        >
          <div
            class="c14"
          >
            <i
              class="far fa-file-alt"
            />
          </div>
          <span>
             messages
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Filters component when fetching matches the snapshot 1`] = `
<body>
  .c9 {
  min-width: 200px;
  height: 32px;
  font-size: 14px;
}

.c9 > .dropdown-container {
  height: 32px;
}

.c9 > .dropdown-container > .dropdown-heading {
  height: 32px;
}

.c3 {
  position: absolute;
  top: 50%;
  line-height: 0;
  z-index: 1;
  left: 12px;
  right: unset;
  height: 11px;
  width: 11px;
  color: #454F54;
}

.c4 {
  border: 1px #ABB5BA solid;
  border-radius: 4px;
  height: 32px;
  width: 100%;
  padding-left: 36px;
  font-size: 14px;
}

.c4::-webkit-input-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c4::-moz-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c4:-ms-input-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c4::placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c4:hover {
  border-color: #73848C;
}

.c4:focus {
  outline: none;
  border-color: #454F54;
}

.c4:focus::-webkit-input-placeholder {
  color: transparent;
}

.c4:focus::-moz-placeholder {
  color: transparent;
}

.c4:focus:-ms-input-placeholder {
  color: transparent;
}

.c4:focus::placeholder {
  color: transparent;
}

.c4:disabled {
  color: #ABB5BA;
  border-color: #E3E6E8;
  cursor: not-allowed;
}

.c4:read-only {
  color: #171A1C;
  border: none;
  background-color: #F1F2F3;
  cursor: not-allowed;
}

.c4:-moz-read-only:focus::placeholder {
  color: #ABB5BA;
}

.c4:read-only:focus::placeholder {
  color: #ABB5BA;
}

.c8 {
  border: 1px #ABB5BA solid;
  border-radius: 4px;
  height: 32px;
  width: 100%;
  padding-left: 12px;
  font-size: 14px;
}

.c8::-webkit-input-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c8::-moz-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c8:-ms-input-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c8::placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c8:hover {
  border-color: #73848C;
}

.c8:focus {
  outline: none;
  border-color: #454F54;
}

.c8:focus::-webkit-input-placeholder {
  color: transparent;
}

.c8:focus::-moz-placeholder {
  color: transparent;
}

.c8:focus:-ms-input-placeholder {
  color: transparent;
}

.c8:focus::placeholder {
  color: transparent;
}

.c8:disabled {
  color: #ABB5BA;
  border-color: #E3E6E8;
  cursor: not-allowed;
}

.c8:read-only {
  color: #171A1C;
  border: none;
  background-color: #F1F2F3;
  cursor: not-allowed;
}

.c8:-moz-read-only:focus::placeholder {
  color: #ABB5BA;
}

.c8:read-only:focus::placeholder {
  color: #ABB5BA;
}

.c2 {
  position: relative;
}

.c6 {
  position: relative;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border: 1px #ABB5BA solid;
  border-radius: 4px;
  font-size: 14px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding-left: 12px;
  padding-right: 16px;
  color: #171A1C;
  min-width: 100px;
  background-image: url('data:image/svg+xml,%3Csvg width="10" height="6" viewBox="0 0 10 6" fill="none" xmlns="http://www.w3.org/2000/svg"%3E%3Cpath d="M1 1L5 5L9 1" stroke="%23454F54"/%3E%3C/svg%3E%0A') !important;
  background-repeat: no-repeat !important;
  background-position-x: calc(100% - 8px) !important;
  background-position-y: 55% !important;
}

.c6:hover {
  color: #171A1C;
  border-color: #73848C;
}

.c6:focus {
  outline: none;
  color: #171A1C;
  border-color: #454F54;
}

.c6:disabled {
  color: #ABB5BA;
  border-color: #E3E6E8;
  cursor: not-allowed;
}

.c11 {
  position: relative;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border: 1px #ABB5BA solid;
  border-radius: 4px;
  font-size: 14px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding-left: 12px;
  padding-right: 16px;
  color: #171A1C;
  min-width: 120px;
  background-image: url('data:image/svg+xml,%3Csvg width="10" height="6" viewBox="0 0 10 6" fill="none" xmlns="http://www.w3.org/2000/svg"%3E%3Cpath d="M1 1L5 5L9 1" stroke="%23454F54"/%3E%3C/svg%3E%0A') !important;
  background-repeat: no-repeat !important;
  background-position-x: calc(100% - 8px) !important;
  background-position-y: 55% !important;
}

.c11:hover {
  color: #171A1C;
  border-color: #73848C;
}

.c11:focus {
  outline: none;
  color: #171A1C;
  border-color: #454F54;
}

.c11:disabled {
  color: #ABB5BA;
  border-color: #E3E6E8;
  cursor: not-allowed;
}

.c7 {
  padding-right: 16px;
  list-style-position: inside;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #F1F2F3;
  color: #171A1C;
  font-size: 14px;
  font-weight: 500;
  height: 32px;
}

.c10:hover:enabled {
  background: #E3E6E8;
  color: #171A1C;
  cursor: pointer;
}

.c10:active:enabled {
  background: #D5DADD;
  color: #171A1C;
}

.c10:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c10 a {
  color: #FFFFFF;
}

.c10 i {
  margin-right: 7px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 16px;
  padding-right: 16px;
}

.c0 > div:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 16px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 90%;
}

.c1 > div:first-child {
  width: 25%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 .select-wrapper {
  width: 40% !important;
}

.c5 .select-wrapper > select {
  border-radius: 4px 0 0 4px !important;
}

.c5 .offset-selector {
  border-radius: 0 4px 4px 0 !important;
  border-left: none;
}

.c5 .date-picker {
  height: 32px;
  border: 1px #ABB5BA solid;
  border-left: none;
  border-radius: 0 4px 4px 0;
  font-size: 14px;
  width: 100%;
  padding-left: 12px;
  color: #171A1C;
  background-image: url('data:image/svg+xml,%3Csvg width="10" height="6" viewBox="0 0 10 6" fill="none" xmlns="http://www.w3.org/2000/svg"%3E%3Cpath d="M1 1L5 5L9 1" stroke="%23454F54"/%3E%3C/svg%3E%0A') !important;
  background-repeat: no-repeat !important;
  background-position-x: 96% !important;
  background-position-y: 55% !important;
  -webkit-appearance: none !important;
  -moz-appearance: none !important;
  appearance: none !important;
}

.c5 .date-picker:hover {
  cursor: pointer;
}

.c5 .date-picker:focus {
  outline: none;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 22px;
  padding-top: 16px;
  padding-bottom: 16px;
}

.c13 {
  color: #73848C;
  font-size: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c14 {
  color: #171A1C;
  padding-right: 6px;
  height: 12px;
}

<div>
    <div
      class="c0"
    >
      <div>
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <i
              class="c3 fas fa-search"
            />
            <input
              class="c4 c2"
              id="searchText"
              placeholder="Search"
              type="text"
              value=""
            />
          </div>
          <div
            class="c5"
          >
            <div>
              <ul
                class="c6"
                id="selectSeekType"
                role="listbox"
              >
                <li
                  class="c7"
                  role="option"
                  tabindex="0"
                >
                  Offset
                </li>
              </ul>
            </div>
            <div
              class="c2 offset-selector"
            >
              <input
                class="c8 c2 offset-selector"
                id="offset"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rmsc c9"
          >
            <div
              aria-labelledby="Select partitions"
              aria-readonly="true"
              class="dropdown-container"
              tabindex="0"
            >
              <div
                class="dropdown-heading"
              >
                <div
                  class="dropdown-heading-value"
                >
                  <span>
                    All items are selected.
                  </span>
                </div>
                <button
                  aria-label="Clear Selected"
                  class="clear-selected-button"
                  type="button"
                >
                  <svg
                    class="dropdown-search-clear-icon gray"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-width="2"
                    width="24"
                  >
                    <line
                      x1="18"
                      x2="6"
                      y1="6"
                      y2="18"
                    />
                    <line
                      x1="6"
                      x2="18"
                      y1="6"
                      y2="18"
                    />
                  </svg>
                </button>
                <svg
                  class="dropdown-heading-dropdown-arrow gray"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-width="2"
                  width="24"
                >
                  <path
                    d="M6 9L12 15 18 9"
                  />
                </svg>
              </div>
            </div>
          </div>
          <button
            class="c10"
            style="font-weight: 500;"
            type="button"
          >
            Cancel
          </button>
        </div>
        <div>
          <ul
            class="c11"
            role="listbox"
          >
            <li
              class="c7"
              role="option"
              tabindex="0"
            >
              Oldest First
            </li>
          </ul>
        </div>
      </div>
      <div
        class="c12"
      >
        <p
          style="font-size: 14px;"
        />
        <div
          class="c13"
          title="Elapsed Time"
        >
          <div
            class="c14"
          >
            <i
              class="far fa-clock"
            />
          </div>
          <span>
            0
             ms
          </span>
        </div>
        <div
          class="c13"
          title="Bytes Consumed"
        >
          <div
            class="c14"
          >
            <i
              class="fas fa-arrow-down"
            />
          </div>
          <span>
            0Bytes
          </span>
        </div>
        <div
          class="c13"
          title="Messages Consumed"
        >
          <div
            class="c14"
          >
            <i
              class="far fa-file-alt"
            />
          </div>
          <span>
             messages
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
